analyses/eth/docs/make_plots.R

```diff
@@ -0,0 +1,107 @@
+library(dplyr)
+library(sf)
+library(readxl)
+library(ggplot2)
+library(tidyr)
+library(tmap)
+
+# Read data ---------------------------------------------------------------
+
+data_dir <- Sys.getenv("AA_DATA_DIR")
+shapefile_path <- paste0(data_dir, "/public/raw/eth/cod_ab/ET_Admin_OCHA_2020")
+
+shp_adm2 <- st_read(paste0(shapefile_path, "/eth_admbnda_adm2_csa_bofed_20201008.shp"))
+shp_adm1 <- st_read(paste0(shapefile_path, "/eth_admbnda_adm1_csa_bofed_20201008.shp"))
+
+processed_country_dir <- paste0(data_dir,"/public/processed/eth/")
+exploration_country_dir <- paste0(data_dir,"/public/exploration/eth/")
+
+emdat <- read_excel(paste0(exploration_country_dir, "emdat_public_2021_10_06_query_uid-vjop9c.xlsx"))
+
+
+# Create plots ------------------------------------------------------------
+
+plot_disaster_annual <- emdat %>%
+  group_by(`Disaster Type`, Year) %>%
+  summarise(n()) %>%
+  mutate(Year = as.numeric(Year)) %>%
+  complete(Year = 1965:2021) %>%
+  mutate(`n()` = replace_na(`n()`, 0)) %>%
+  ggplot(aes(x=Year, y=`n()`, group=`Disaster Type`, color =`Disaster Type`)) +
+    geom_line() +
+    theme_minimal() +
+    labs(x='Year', y='Number') +
+    theme(legend.position="bottom")
+
+
+
+# Get number of events by admin 1 
+# TODO: Fuzzy matching in case slight mismatch in spelling
+# Join with events geolocated to adm2 as well
+
+df_count <- data.frame(adm1 = shp_adm1$ADM1_EN)
+df_count['num_flood'] <- 0
+df_count['num_drought'] <- 0
+
+emdat_flood <- emdat %>% filter(`Disaster Type` == 'Flood')
+emdat_drought <- emdat %>% filter(`Disaster Type` == 'Drought')
+
+# FLOOD
+for (adm1 in shp_adm1$ADM1_EN) {
+  for (event_loc in emdat_flood$Location){
+    match <- grepl(adm1, event_loc, fixed=TRUE)
```

Comment:
Are we positive there are only perfect matches and we don't need to be concerned with fuzzy ones?

Reply:
No, not positive! I definitely still need to go through these with a finer-tooth comb and do some QA for this matching. I also need to find a way to address the fact that not all events are consistently located to the admin 1 level - the location references frustratingly aren't standardized.

Reply:
With `grepl()` you aren't doing any fuzzy matching, it's just exact matching. When quickly checking I noticed for instance `Gambella` wasn't being captured (or `Gambela`, whichever is *not* the official name). You could try using `agrepl()` for fuzzy matching. If using the `match_fun` example put above, it could look something like:

```r
match_fun = ~ agrepl(
    paste0("\\b", .x, "\\b"),
    .y,
    max_distance = 1L,
    fixed = F,
    ignore.case = T
)
```

That should fuzzy match only allowing a single letter difference between the words.